### ISSUE TITLE
Create waypoints with map

### ISSUE THREAD

**Issue by awohns** [awohns/tspyro#7]
I'm trying to use an image to mask the [hex grid function for creating waypoints](https://github.com/awohns/tspyro/blob/68c11110d7abdce2bb9f424cbfa1e01ace0ef4f4/tspyro/diffusion.py#L194). However, [my code isn’t properly dealing with the fact that the hex grid has rows which alternate between the edges of the hexagons](https://github.com/awohns/tspyro/blob/68c11110d7abdce2bb9f424cbfa1e01ace0ef4f4/tspyro/diffusion.py#L230).
```python
from PIL import Image
import torchvision
import numpy as np
import torch

from tspyro.diffusion import make_hex_grid

import matplotlib.pyplot as plt

img = Image.open('world_map_540x217.png')
img = img.convert('L')
pixels = np.asarray(img)
coords = pixels < 1
coords = torch.from_numpy(coords)
# coords = np.array([coords[:,1], coords[:,0]])
coords = coords.unsqueeze(0)

def on_land(x, y):
    scaled_mask = torchvision.transforms.functional.resize(
        coords, size=(x.shape[0], x.shape[1])).numpy()
    return scaled_mask


bounds = dict(west=0, east=img.size[0], south=0, north=img.size[1])
grid = make_hex_grid(**bounds, radius=5, predicate=on_land)
x = grid["waypoints"][:,0]
y = grid["waypoints"][:,1]
plt.scatter(x, y.max() - y)
```
![Screen Shot 2021-11-16 at 4 14 50 PM](https://user-images.githubusercontent.com/10664424/142066789-e7c65642-528f-4c38-b4f9-b1dda7c1ff14.png)

You can see in the output that there’s some weird doubling of the southern hemisphere. However, if I remove the dx, dy part of the loop in the hex grid creation (see below), things work as expected when rerunning the first code block:
```python
for x0, y0 in [(west, south)]:
        X = torch.arange(1 + math.floor((east - west) / dx)) * dx + x0
        Y = torch.arange(1 + math.floor((north - south) / dy)) * dy + y0
        X, Y = torch.broadcast_tensors(X, Y[:, None])
        parts.append(torch.stack([X, Y], dim=-1))
    waypoints = torch.cat(parts, dim=0)
```

![Screen Shot 2021-11-16 at 4 14 29 PM](https://user-images.githubusercontent.com/10664424/142066721-cb40e02a-533a-4036-b1fa-c2ade361b5f1.png)

I tried halving the 1st dimension of the image in the torchvision reshape, and then using `torch.repeat_interleave()`, but this didn't seem to fix the issue.

Any thoughts on how this could be fixed @fritzo, @eb8680?

**Comment by fritzo** [comment]
You'll need to change the `on_land(-,-)` function. It looks like you were resizing the image based on the hex grid, but instead you should use the `(x,y)` waypoints to index into the image. Here's an example:
```py
img = Image.open('world_map_540x217.png')
img = img.convert('L')
pixels = np.asarray(img)
coords = pixels < 1
coords = torch.from_numpy(coords)
assert coords.dim() == 2

def on_land(x, y):
    return coords[x.round().long(), y.round().long()]  # just index into the image
```
